### ISSUE TITLE
Upgrade React and react-test-render to 18

### ISSUE THREAD

**Issue by acezard** [cozy/cozy-flagship-app#577]
And more generally all react-dependent libraries to react 18?

First, verify if it's even possible with our current react native version

**Comment by Ldoppea** [comment]
I'm not sure that we should deviate from the official react's version. But react-native 0.69 has react 18 support ([cf](https://reactnative.dev/docs/react-18-and-react-native)). So I would prefer to upgrade react-native instead.

**Comment by acezard** [comment]
> I'm not sure that we should deviate from the official react's version. But react-native 0.69 has react 18 support ([cf](https://reactnative.dev/docs/react-18-and-react-native)). So I would prefer to upgrade react-native instead.

Yeah absolutely, I wasn't sure we could actually upgrade to 18 so easily. I guess this issue will be dependent on upgrading RN then (we went a long time without upgrading)
